plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestingIcebergHiveMetastoreCatalogModule.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.hms;
+
+import com.google.inject.Binder;
+import com.google.inject.Scopes;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.plugin.hive.metastore.DecoratedHiveMetastoreModule;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.HiveMetastoreFactory;
+import io.trino.plugin.hive.metastore.RawHiveMetastoreFactory;
+import io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastore;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastoreFactory;
+import io.trino.plugin.iceberg.catalog.IcebergTableOperationsProvider;
+import io.trino.plugin.iceberg.catalog.TrinoCatalogFactory;
+
+import static java.util.Objects.requireNonNull;
+
+public class TestingIcebergHiveMetastoreCatalogModule
+        extends AbstractConfigurationAwareModule
+{
+    private final HiveMetastore metastore;
+    private final ThriftMetastore thriftMetastore;
+
+    public TestingIcebergHiveMetastoreCatalogModule(ThriftMetastore thriftMetastore)
+    {
+        this.thriftMetastore = requireNonNull(thriftMetastore, "thriftMetastore is null");
+        this.metastore = new BridgingHiveMetastore(thriftMetastore);
+    }
+
+    public HiveMetastore getHiveMetastore()
+    {
+        return metastore;
+    }
+
+    @Override
+    protected void setup(Binder binder)
+    {
+        install(new DecoratedHiveMetastoreModule());
+        binder.bind(ThriftMetastoreFactory.class).toInstance(ThriftMetastoreFactory.ofInstance(thriftMetastore));
+        binder.bind(HiveMetastoreFactory.class).annotatedWith(RawHiveMetastoreFactory.class).toInstance(HiveMetastoreFactory.ofInstance(metastore));
```

Comment:
create an instance on the fly instead of adding code on `HiveMetastoreFactory` (I refer to `HiveMetastoreFactory.ofInstance` method).